fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,281 @@
+
+
+def _convert_scalar_error_to_array_error(df_errors):
+    if all((isinstance(error, float) or isinstance(error, int)) for error in df_errors):
+        return df_errors.apply(lambda x: [x, x])
+    if all((_is_arraylike(error) and len(error) == 2) for error in df_errors):
+        return df_errors
+    else:
+        raise ValueError(_ERRORS_MUST_BE_SCALAR_OR_ARRAY)
+
+
+def _check_valid_errors(df_errors):
+    for tup in df_errors:
+        if not all(ele >= 0 for ele in tup):
+            raise ValueError(_ERRORS_NEGATIVE_VALUE_ERROR)
+
+
+def _check_valid_conf_interval(df_conf_intervals):
+    for tup in df_conf_intervals:
+        if not _is_arraylike(tup) or len(tup) != 2:
+            raise ValueError(_CONF_INTERVALS_MUST_BE_ARRAY)
+        if tup[0] > tup[1]:
+            raise ValueError(_CONF_INTERVALS_FLIPPED_BOUNDS_ERROR)
+
+
+def _build_legend(ax, kind, legend_label):
+    color = ax.lines[0].get_color() if kind == "scatter" else "black"
+
+    # extend legend with 95% CI text
+    handles, labels = ax.get_legend_handles_labels()
+    custom_line = [Line2D([0], [0], color=color, label=legend_label)]
+    handles.extend(custom_line)
+    ax.legend(handles=handles)
+
+
+def _plot_df(df, metrics, kind, subplots, legend_label, df_all_errors=None, **kwargs):
+    if df_all_errors is not None:
+        yerr = np.array(
+            [np.array([[*row] for row in df_all_errors[metric]]).T for metric in metrics]
+        )
+        if kind == "scatter":
+            axs = df[metrics].plot(linestyle='', marker='o', yerr=yerr,
+                                   subplots=subplots, **kwargs)
+        else:
+            axs = df[metrics].plot(kind=kind, yerr=yerr, subplots=subplots, **kwargs)
+
+        if isinstance(axs, np.ndarray):
+            for ax in axs.flatten():
+                _build_legend(ax, kind, legend_label)
+        else:
+            _build_legend(axs, kind, legend_label)
+
+    else:
+        if kind == "scatter":
+            axs = df[metrics].plot(linestyle='', marker='o', subplots=subplots, **kwargs)
+        else:
+            axs = df[metrics].plot(kind=kind, subplots=subplots, **kwargs)
+
+    return axs
+
+
+def plot_metric_frame(metric_frame: MetricFrame, *,
+                      kind: str = "scatter",
+                      metrics: Union[List[str], str] = None,
+                      errors: Union[List[str], str] = None,
+                      conf_intervals: Union[List[str], str] = None,
+                      subplots: bool = True,
+                      plot_error_labels: bool = False,
+                      error_labels_precision: int = 4,
+                      error_labels_fontsize: int = 8,
+                      error_labels_color: str = "black",
+                      error_labels_ha: str = "center",
+                      legend_label: str = "95% CI",
+                      **kwargs
+                      ) -> Union[Axes, List[Axes]]:
+    """Visualization for metrics with statistical error bounds.
```

Comment:
This works also w/o CI, so it's not only with statistical error bounds. Also, the error bounds don't have to be "statistical", they could be whatever the user decides them to be.